epoxy-sample/src/main/java/com/airbnb/epoxy/sample/models/ColorModel.java

```diff
@@ -54,6 +61,31 @@ public void bindChanges(@NonNull ColorHolder holder,
     }
   }
 
+  @Nullable
+  @Override
+  public EpoxyItemAnimation bindChangesAnimated(@NonNull ColorHolder view,
+                                                @NonNull EpoxyModel<?> previouslyBoundModel,
+                                                @Nullable EpoxyItemAnimation interruptedAnimation) {
+
+    int currentColor = ((ColorModel) previouslyBoundModel).color;
+    float currentRotation = 0f;
+
+    if(interruptedAnimation != null) {
+      AnimatorSet previousAnim = (AnimatorSet) ((EpoxyItemAnimation.AnimatorWrapper) interruptedAnimation).wrappedAnimator;
```

Comment:
I see. Well, Epoxy is API 16+ so people won't need `ViewPropertyAnimatorCompat`. It would be nice to be able to just use `Animator` to simplify things, but I'm not sure how much it would limit people. Since `EpoxyItemAnimation` only supports Animator by default, I'm not sure how many people would actually use it for other things like Lottie since it would be a decent about of extra work to create the wrapper. I imagine in that case they would just manage the animation directly via lottie like I was doing in the sample before.

I think the `previousModel` is useful for knowing which animations to run - eg a "like" animation vs a "vote" animation - if the view supports different animations. Also, what if something changes that they don't want to animate. Maybe they want to animate a "like" change but don't want to animate an image change.

You're right that users could manage animations themselves. I think that removing the overhead of storing the animation and remembering to stop it correctly would be great things to handle automatically though. Otherwise I imagine people will make mistakes doing it themselves. Does the item animator also use the change animation duration to coordinate timing with move and removal animations?

The simplest thing I can think of it have `bindChanges` run during `Adapter.bindViewHolder` and return a Nullable `Animator`. We can store that in the epoxy viewholder and use it in the item animator in `animateChange`. That is quite a change from convention, and a little confusing in it's own right.

I'm thinking that it's a fundamental problem that binding a view and animating changes happen at different times, but are changing the same data. Combining them into one method would make sense in some ways.